[CI] ClientYamlTestSuiteIT test {yaml=tsdb/30_snapshot/Create a snapshot and then restore it} failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/3ugnjqj5jxlmu/tests/:rest-api-spec:yamlRestTest/org.elasticsearch.test.rest.ClientYamlTestSuiteIT/test%20%7Byaml=tsdb%2F30_snapshot%2FCreate%20a%20snapshot%20and%20then%20restore%20it%7D


**Reproduction line:**
```
./gradlew ':rest-api-spec:yamlRestTest' --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT.test {yaml=tsdb/30_snapshot/Create a snapshot and then restore it}" -Dtests.seed=B53AAFBDC8D992E8 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=es-PY -Dtests.timezone=Iceland -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
[Failure dashboard for `org.elasticsearch.test.rest.ClientYamlTestSuiteIT#test {yaml=tsdb/30_snapshot/Create a snapshot and then restore it}`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Byaml%3Dtsdb/30_snapshot/Create%20a%20snapshot%20and%20then%20restore%20it%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.test.rest.ClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: Failure at [tsdb/30_snapshot:127]: expected [2xx] status code but api [snapshot.restore] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"null_pointer_exception","reason":"Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null","stack_trace":"org.elasticsearch.ElasticsearchException$1: Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:704)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:632)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.rest.RestResponse.build(RestResponse.java:184)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:143)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:114)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:53)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:214)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:234)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:186)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:191)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:247)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.Repository.lambda$getSnapshotInfo$1(Repository.java:103)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$getSnapshotInfo$25(BlobStoreRepository.java:1813)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.GetSnapshotInfoContext.onResponse(GetSnapshotInfoContext.java:110)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$getOneSnapshotInfo$26(BlobStoreRepository.java:1864)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1570)\nCaused by: java.lang.NullPointerException: Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null\n\tat java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:337)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.common.util.set.Sets.union(Sets.java:134)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.snapshots.RestoreService.startRestore(RestoreService.java:407)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.snapshots.RestoreService.lambda$restoreSnapshot$3(RestoreService.java:288)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)\n\t... 8 more\n"}],"type":"null_pointer_exception","reason":"Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null","stack_trace":"java.lang.NullPointerException: Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null\n\tat java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:337)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.common.util.set.Sets.union(Sets.java:134)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.snapshots.RestoreService.startRestore(RestoreService.java:407)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.snapshots.RestoreService.lambda$restoreSnapshot$3(RestoreService.java:288)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.Repository.lambda$getSnapshotInfo$1(Repository.java:103)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$getSnapshotInfo$25(BlobStoreRepository.java:1813)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.GetSnapshotInfoContext.onResponse(GetSnapshotInfoContext.java:110)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$getOneSnapshotInfo$26(BlobStoreRepository.java:1864)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1570)\n"},"status":500}]

  at __randomizedtesting.SeedInfo.seed([B53AAFBDC8D992E8:3D6E90676625FF10]:0)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:610)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:553)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1570)

  Caused by: java.lang.AssertionError: expected [2xx] status code but api [snapshot.restore] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"null_pointer_exception","reason":"Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null","stack_trace":"org.elasticsearch.ElasticsearchException$1: Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:704)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:632)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.rest.RestResponse.build(RestResponse.java:184)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:143)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:114)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:53)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:214)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:234)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:186)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:191)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:247)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.Repository.lambda$getSnapshotInfo$1(Repository.java:103)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$getSnapshotInfo$25(BlobStoreRepository.java:1813)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.GetSnapshotInfoContext.onResponse(GetSnapshotInfoContext.java:110)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$getOneSnapshotInfo$26(BlobStoreRepository.java:1864)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1570)\nCaused by: java.lang.NullPointerException: Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null\n\tat java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:337)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.common.util.set.Sets.union(Sets.java:134)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.snapshots.RestoreService.startRestore(RestoreService.java:407)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.snapshots.RestoreService.lambda$restoreSnapshot$3(RestoreService.java:288)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)\n\t... 8 more\n"}],"type":"null_pointer_exception","reason":"Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null","stack_trace":"java.lang.NullPointerException: Cannot invoke \"java.util.Collection.iterator()\" because \"c\" is null\n\tat java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:337)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.common.util.set.Sets.union(Sets.java:134)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.snapshots.RestoreService.startRestore(RestoreService.java:407)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.snapshots.RestoreService.lambda$restoreSnapshot$3(RestoreService.java:288)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.Repository.lambda$getSnapshotInfo$1(Repository.java:103)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$getSnapshotInfo$25(BlobStoreRepository.java:1813)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.GetSnapshotInfoContext.onResponse(GetSnapshotInfoContext.java:110)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$getOneSnapshotInfo$26(BlobStoreRepository.java:1864)\n\tat org.elasticsearch.server@8.15.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1570)\n"},"status":500}]

    at org.junit.Assert.fail(Assert.java:89)
    at org.elasticsearch.test.rest.yaml.section.DoSection.checkResponseException(DoSection.java:540)
    at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:396)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:590)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:553)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke(Method.java:580)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:1570)

```

--- a/server/src/main/java/org/elasticsearch/snapshots/RestoreService.java
+++ b/server/src/main/java/org/elasticsearch/snapshots/RestoreService.java
@@ -404,9 +404,9 @@ public final class RestoreService implements ClusterStateApplier {
                     .flatMap(ds -> ds.getFailureIndices().getIndices().stream().map(idx -> new Tuple<>(ds.isSystem(), idx.getName())))
                     .collect(Collectors.partitioningBy(Tuple::v1, Collectors.mapping(Tuple::v2, Collectors.toSet())));
             }
-            systemDataStreamIndices = Sets.union(backingIndices.get(true), failureIndices.get(true));
-            nonSystemDataStreamBackingIndices = backingIndices.get(false);
-            nonSystemDataStreamFailureIndices = failureIndices.get(false);
+            systemDataStreamIndices = Sets.union(backingIndices.getOrDefault(true, Set.of()), failureIndices.getOrDefault(true, Set.of()));
+            nonSystemDataStreamBackingIndices = backingIndices.getOrDefault(false, Set.of());
+            nonSystemDataStreamFailureIndices = failureIndices.getOrDefault(false, Set.of());
         }
         requestIndices.addAll(nonSystemDataStreamBackingIndices);
         requestIndices.addAll(nonSystemDataStreamFailureIndices);

